remme/ws/events.py

```diff
@@ -0,0 +1,103 @@
+import json
+
+import zmq
+import logging
+import weakref
+import asyncio
+
+from remme.settings import ZMQ_URL
+from sawtooth_sdk.protobuf.client_event_pb2 import ClientEventsSubscribeRequest
+from sawtooth_sdk.protobuf.validator_pb2 import Message
+from sawtooth_sdk.protobuf.events_pb2 import EventList, EventSubscription
+from google.protobuf.json_format import MessageToJson
+
+from remme.shared.utils import generate_random_key
+from remme.ws.basic import BasicWebSocketHandler
+from remme.ws.constants import Entity
+
+SWAP_INIT_EVENT = 'atomic-swap/init'
+
+LOGGER = logging.getLogger(__name__)
+
+
+class WSEventSocketHandler(BasicWebSocketHandler):
+    def __init__(self, stream, loop):
+        super().__init__(stream, loop)
+        self._events = [SWAP_INIT_EVENT]
+        self._events_updator_task = weakref.ref(
+            asyncio.ensure_future(
+                self.listen_events(), loop=self._loop))
+        self.subscribe_events()
+
+    # return what value to be mapped to web_sock
+    def subscribe(self, entity, data):
+        if entity == Entity.EVENTS:
+            events = data.get('events', [])
+            return {'events': events}
+
+    def unsubscribe(self, entity, data):
+        pass
+
+    def subscribe_events(self):
+        # Setup a connection to the validator
+        LOGGER.info(f"Subscribing to events")
+        ctx = zmq.Context()
```

Comment:
Why not in __init__section?
Why need to use Context instead `from sawtooth_sdk.messaging.stream import Stream` ?

Reply:
https://github.com/Remmeauth/remme-core/pull/64